lib/salesforce/api.rb

```diff
@@ -9,29 +9,37 @@ class InvalidCredentials < ::StandardError; end
 
   class Api
 
-    VERSION = "49.0".freeze
-    INVALID_RESPONSE = "salesforce.error.invalid_response".freeze
+    VERSION = "49.0"
+    INVALID_RESPONSE = "salesforce.error.invalid_response"
+    APP_NOT_APPROVED = "dashboard.salesforce.app_not_approved"
 
-    attr_reader :faraday, :path
+    attr_reader :faraday, :prefix
 
     def initialize
       set_access_token
 
       @faraday = Faraday.new(
         url: SiteSetting.salesforce_instance_url,
-        headers: { 'Authorization' => "Bearer #{SiteSetting.salesforce_access_token}" }
+        headers: { 'Authorization' => "Bearer #{access_token}" }
       )
-      @path = "/services/data/v#{VERSION}"
+      @prefix = "/services/data/v#{VERSION}"
     end
 
-    def get(uri)
-      uri = File.join(path, uri)
-      response = faraday.get(uri)
+    def get(path)
+      call(path) do |uri|
+        faraday.get(uri)
```

Comment:
I think we should consider what happens when the request does not return a 200 response code too.